### ISSUE TITLE
Font is blurry on MacOS

### ISSUE THREAD

**Issue by aslpavel** [jwilm/alacritty#486]
Fonts look considerably more blurry if compare them with iTerm. It is especially noticeable for blod variant.

![screen shot 2017-03-17 at 12 02 27](https://cloud.githubusercontent.com/assets/281294/24042420/c4dc62ea-0b09-11e7-8b34-8d4545f216dc.png)

**Comment by okuuva** [comment]
Have you tried setting `use_thin_strokes: true` under `font` section in alacritty.yml?

**Comment by aslpavel** [comment]
Yes this result screenshot taken with `use_thin_strokes: true`.
Oddly when I use dark background result look way better.
__dark__
![screen shot 2017-03-29 at 15 03 41](https://cloud.githubusercontent.com/assets/281294/24458550/26799078-1491-11e7-9217-33b94f29dd51.png)
 __light__
![screen shot 2017-03-29 at 15 03 16](https://cloud.githubusercontent.com/assets/281294/24458570/32cebd44-1491-11e7-98c4-682d65eb563d.png)

So it is just a guess, but how do we cache glyphs, it might be we use the same cached glyph for different background/foreground colors, it might explain the difference here.

**Comment by hlieberman** [comment]
This is probably anti-aliasing related.  The contrast on dark backgrounds is what's making it look better, rather than it actually be rendering differently.

**Comment by matttproud** [comment]
I am no rendering expert, but I am relatively certain that it is an interaction of [Core Graphics](https://developer.apple.com/reference/coregraphics) anti-aliasing and smoothing parameters [as specified](https://github.com/jwilm/alacritty/blob/master/font/src/darwin/mod.rs#L412-L413).  I use [Go Font](https://blog.golang.org/go-fonts) on MacOS, and it rendered poorly on a light background until I set those aforementioned smoothing directives to `false`.

**Comment by aslpavel** [comment]
This might be related [iterm2 issue](https://gitlab.com/gnachman/iterm2/issues/4986), they are enabling font smoothing based on brighness

**Comment by mystal** [comment]
I'm also noticing in @aslpavel's screenshots that the colors are brighter compared to iterm2. I've seen that same effect locally and am wondering what the cause could be...

**Comment by evmorov** [comment]
@aslpavel have you found a solution the improve the font in Alacritty? I have the same issue.

**Comment by aslpavel** [comment]
No issue has not been resolved

**Comment by tpict** [comment]
![untitled-1](https://user-images.githubusercontent.com/12588098/36846680-7fc11040-1d53-11e8-9b25-ad92b9efc6c2.png)
Here's an example with Alacritty on top, iTerm2 metal renderer beneath. Looks like there's something wonky going on with the hinting, round letters all look very uneven.

**Comment by krlvi** [comment]
One workaround for disabling core graphics subpixel AA etc until RFC https://github.com/jwilm/alacritty/issues/957 is out is to hardcode these to false before compiling the app:
https://github.com/jwilm/alacritty/blob/1adb5cb7fc05054197aa08e0d1fa957df94888ab/font/src/darwin/mod.rs#L532-L537
The weird blur goes away

**Comment by chrisniael** [comment]
As far as I know, MacOS Mojave disables subpixel antialiasing, also known as font smoothing, by default. 
Maybe this will helps someone to resolve the problem.
Btw, @krlvi 's solution makes sense to me in my MacBook Pro.

https://github.com/jwilm/alacritty/blob/1adb5cb7fc05054197aa08e0d1fa957df94888ab/font/src/darwin/mod.rs#L528-L533

And I just set `smoothing` be `false`, that makes me feel better.

```rust
        if use_thin_strokes {
            //cg_context.set_font_smoothing_style(16);
        }

        cg_context.set_allows_font_smoothing(false);
        cg_context.set_should_smooth_fonts(false);
```

**Comment by russelldavies** [comment]
What @chrisniael suggests works but makes the strokes a bit too thin compared to iTerm2 in my opinion.

Default:
![default](https://user-images.githubusercontent.com/120115/51442980-cbd9af00-1cda-11e9-8eb0-6bcae802b6f2.png)

Smoothing disabled:
![custom](https://user-images.githubusercontent.com/120115/51442981-d300bd00-1cda-11e9-9e55-40b2b67ede2e.png)

iTerm2:
![iterm](https://user-images.githubusercontent.com/120115/51442998-fd527a80-1cda-11e9-935f-f5e645896770.png)

**Comment by russelldavies** [comment]
@chrisduerr I think this can also be closed since it mirrors #1635.

**Comment by chrisduerr** [comment]
Thanks!
